tests/net/socket/tcp/src/main.c

```diff
@@ -709,18 +709,29 @@ void test_open_close_immediately(void)
 	zassert_not_equal(connect(c_sock, (struct sockaddr *)&s_saddr,
 				  sizeof(s_saddr)),
 			  0, "connect succeed");
+
 	test_close(c_sock);
 
+	/* Allow for the close communication to finish,
+	 * this makes the test success, no longer scheduling dependent
+	 */
+	k_sleep(K_MSEC(CONFIG_NET_TCP_INIT_RETRANSMISSION_TIMEOUT/2));
```

Comment:
I would suggest that you set the `CONFIG_NET_TCP_INIT_RETRANSMISSION_TIMEOUT` in the test config file just in case the default value changes. This way we could avoid long timeouts etc hard to debug issues later (not likely but still on option).

Reply:
Totally agree, fixed